java/ql/src/experimental/Security/CWE/CWE-555/CredentialsInPropertiesFile.qhelp

```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+  <overview>
+    <p>
+      Credentials management issues occur when credentials are stored in plaintext in 
+      an application’s properties file. Common credentials include but are not limited 
+      to LDAP, mail, database, proxy account, and so on. Storing plaintext credentials 
+      in a properties file allows anyone who can read the file access to the protected 
+      resource. Good credentials management guidelines require that credentials never 
+      be stored in plaintext.
+    </p>
+  </overview>
+
+  <recommendation>
+    <p>
+      Credentials stored in properties files should be encrypted and recycled regularly. 
+      In a Java EE deployment scenario, utilities provided by application servers like 
+      keystores and password vaults can be used to encrypt and manage credentials.
+    </p>
+  </recommendation>
+
+  <example>
+    <p>
+      In the first example, the credentials of a LDAP and datasource properties are stored 
+      in cleartext in the properties file.
+    </p>
+
+    <p>
+      In the second example, the credentials of a LDAP and datasource properties are stored
```

Comment:
```suggestion
      In the first example, the credentials for the LDAP and datasource are stored 
      in cleartext in the properties file.
    </p>

    <p>
      In the second example, the credentials for the LDAP and datasource are stored
```
The second "properties" sounds confusing. Additionally LDAP stands for "Lightweight Directory Access Protocol" so using "a LDAP" sounds weird since there is only one LDAP.

Reply:
Changed